### PR TITLE
docs: Document supported `consul connect` env vars

### PR DESCRIPTION
**Pull request by blake**
Document the ability to specify `-sidecar-for` and `-proxy-id` flags via environment variables.

These environment variables have been supported for `consul connect proxy` since Connect was introduced in 1.2.0.

Support was added for using these variables with `consul connect envoy` in Consul 1.8.0 with PR #7498.

### CODE REVIEW

**Review by freddygv — APPROVED** [review]
Really minor but I noticed the disclaimer is kind of inconsistently placed. For proxy-id it shows up at the end of the paragraph after the note about the service ID needing to be registered on the agent. For sidecar-for the note shows up before the comment about the target not needing to exist on the agent.

Will leave it up to you if we want to make the ordering consistent.

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/419817.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 17e18df81e3f4bf547c0bc1aeba69188c44d1e51 onto `stable-website` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 17e18df81e3f4bf547c0bc1aeba69188c44d1e51 onto `release/1.10.x` succeeded!

**Pull request merged**
